### PR TITLE
Docs: Add missing visualizations to Grafana vizualization index page

### PR DESCRIPTION
**Pull request by srperf**
There were 3 visualization types missing in the list. Added: Geomap, Canvas and Gauge.
Hope the links work, just added them at the bottom, although Gauge was already there.

**What is this feature?**

Noticed there were some visualizations missing in the list.
I am thinking for a future PR to sort the visualizations alphabetically for ease when trying to find them instead of the weird sections: Misc, Stats, Graphs, etc. This would match the left menu as well, although for some reason Time series is not alphabetical.

**Why do we need this feature?**

The items are missing in the doc.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:
No issue open.

**Special notes for your reviewer:**
I just added the links at the bottom. I hope they work.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by imatwawana — COMMENTED** [review]
Thanks for the update! I definitely want the DataViz squad to look at the first description. I've also updated the entries to have them conform to the same pattern of the other descriptions in the list.

**Comment by imatwawana** [comment]
To answer your question about why Time series isn't alphabetical, it's because it's the most visited visualization, so in consultation with the team, we decided to leave it at the top of the list. I think there's value in leaving the list on this page grouped by visualization type/use case because we don't have another way of conveying that type of grouping easily. If users are trying to navigate to a page, they always have the menu, which is basically alphabetical.

**Review by nmarrs — APPROVED** [review]
Hi @srperf! Thank you for this contribution :) 

I think these changes make sense, I'll apply @imatwawana's suggestions and we can get this merged 🎸

File: docs/sources/panels-visualizations/visualizations/_index.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ If you are unsure which visualization to pick, Grafana can provide visualization
   - [Heatmap][] visualizes data in two dimensions, used typically for the magnitude of a phenomenon.
   - [Pie chart][] is typically used where proportionality is important.
   - [Candlestick][] is typically for financial data where the focus is price/data movement.
+  - [Gauge][] is the traditional rounded visual showing how far a single metric is from a threshold.
```
Comment (nmarrs) [inline docs/sources/panels-visualizations/visualizations/_index.md:36]:
```suggestion
  - [Gauge][] is used to display a single value of data in a quantitative way.
```
cc @imatwawana thoughts on this rewrite?

Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/_index.md:36]: I don't find the rewrite easier to understand. If you couldn't use the words "in a quantitative way", what would you say instead?

Comment (nmarrs) [reply docs/sources/panels-visualizations/visualizations/_index.md:36]: writing is hard 😬 we can keep your version 😅

Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/_index.md:36]: LOL! I struggle with any words that are math-adjacent, so it's partly me.

**Pull request merged**